proposals/params-span.md

```diff
@@ -21,125 +22,156 @@ public static class Console
+Overload resolution prefers overloads that are applicable in [_normal form_](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#11642-applicable-function-member) over [_expanded form_](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#11642-applicable-function-member).
 
-For overloads that are applicable in _expanded form_, better function member ([§11.6.4.3](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#11643-better-function-member)) will be updated to prefer `params` types in a specific order:
+[_Better function member_](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#11643-better-function-member) will prefer `params ReadOnlySpan<T>` over `params T[]` for overloads applicable in [_expanded form_](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#11642-applicable-function-member):
 
-> When performing this evaluation, if `Mp` or `Mq` is applicable in its expanded form, then `Px` or `Qx` refers to a parameter in the expanded form of the parameter list.
-> 
 > In case the parameter type sequences `{P1, P2, ..., Pn}` and `{Q1, Q2, ..., Qn}` are equivalent (i.e. each `Pi` has an identity conversion to the corresponding `Qi`), the following tie-breaking rules are applied, in order, to determine the better function member.
 > 
 > *  If `Mp` is a non-generic method and `Mq` is a generic method, then `Mp` is better than `Mq`.
 > *  ...
 > *  **Otherwise, if both methods have `params` parameters and are applicable only in their expanded forms, and the `params` types are distinct types with equivalent element type (there is an identity conversion between element types), the more specific `params` type is the first of:**
 >    *  **`ReadOnlySpan<T>`**
->    *  **`Span<T>`**
 >    *  **`T[]`**
->    *  **`IEnumerable<T>`**
 > *  Otherwise if one member is a non-lifted operator and  the other is a lifted operator, the non-lifted one is better.
 > *  Otherwise, neither function member is better.
 
-### Array creation expressions
-Array creation expressions that are target-typed to `ReadOnlySpan<T>` or `Span<T>` will be created on the stack _if the length of the array is a constant value within limits (if any) set by the compiler_.
-Otherwise the array will be allocated on the heap.
+### Array allocation and reuse
+The compiler will include the following optimizations for implicitly allocated `params` arrays. Additional optimizations may be added in future for cases where the compiler can determine there are _no reachable aliases_ to the array.
 
-```csharp
-Span<int> s = new[] { i, j, k };   // int[] on the stack
-WriteLine(fmt, new[] { x, y, z }); // object[] on the stack for WriteLine(string fmt, ReadOnlySpan<object> args);
-```
+The compiler _will allocate the array on the stack_ for a `params ReadOnlySpan<T>` argument when
+- the parameter is implicitly or explicitly `scoped`,
```

Comment:
Could you remind me why we won't allocate on the stack when the parameter is marked unscoped? Is it because we think we can't reuse the array across calls, etc. in that case?

Reply:
Yes, the parameter must be `scoped` to allow the array to be reused. I've added that here, thanks.